### ISSUE TITLE
Remote hypervisor: pod fails to start with io.katacontainers.config.hypervisor.virtio_fs_extra_args annotation

### ISSUE THREAD

**Issue by RobertKrawitz** [kata-containers/kata-containers#6345]
# Description of problem

OpenShift 4.12, kata containers 2.5.2-1.el8.x86_64

Attempted to start the following pod:
```
apiVersion: v1
kind: Pod
metadata:
  name: hello-openshift
  labels:
    app: hello-openshift
  annotations:
    io.katacontainers.config.hypervisor.virtio_fs_extra_args: "[ \"-o\", \"allow_direct_io\" ]"
spec:
  runtimeClassName: kata-remote-cc
  imagePullPolicy: IfNotPresent
  containers:
    - name: hello-openshift
      image: quay.io/rkrawitz/bench-army-knife:latest
      command:
        - "/bin/sleep"
      args:
        - "infinity"
```

# Expected result

Pod should start.

# Actual result

Pod fails to start with the following error:
```
  Warning  FailedCreatePodSandBox  3m14s (x25 over 4m10s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = CreateContainer failed: annotation io.katacontainers.config.hypervisor.virtio_fs_extra_args is not enabled: unknown
```

# Further information

Pod starts with standard (runc) runtime and with kata

**Comment by bpradipt** [comment]
cc @yoheiueda @stevenhorsman

**Comment by stevenhorsman** [comment]
So just to add clarity and check I understand this - in remote hypervisor I don't think we can support virtio fs, so I think the bug here is that rather than just failing to create the pod we want to either ignore the annotation that we can't deal with, or at least give a warning to say it isn't supported, rather than just failing. I've not looked into the issue in any detail yet though.

**Comment by c3d** [comment]
@RobertKrawitz Can you confirm that the pod starts if you remove the annotation?

@stevenhorsman I'm not even sure that we should consider this a bug. The annotation here is for performance, so dropping it might be OK in that case, but the same mechanism can be used for functional annotations, e.g. how you deal with extended attributes, and in that case, the filesystem will behave differently. Since it's relatively easy for the user to remove the annotation, I think that the best course of action is to document the limitation.

@bpradipt Do you agree?

**Comment by RobertKrawitz** [comment]
It starts if I remove the annotation.

@c3d Pods start if other random annotations are present.  I think it would be easiest if unknown annotations were ignored.

**Comment by c3d** [comment]
> It starts if I remove the annotation.
> 
> @c3d Pods start if other random annotations are present. I think it would be easiest if unknown annotations were ignored.

It is expected that you ignore annotations that are not for you. However, I suspect that you would have a similar problem if you used another annotation that the kata runtime actually consumes. For example, I expect a similar error if you use `io.katacontainers.config.hypervisor.path` or `io.katacontainers.config.hypervisor.image`. And as well it should, if we cannot support that on a remote hypervisor.

**Comment by ariel-adam** [comment]
This issue is currently mapped to release 0.6.0 schedule for the 31th of May. 
Is this feasible or should we move it to the next release?

**Comment by yoheiueda** [comment]
Currently, containerd for peer pods is configured to pass annotations as follows.

https://github.com/confidential-containers/cloud-api-adaptor/blob/2536ae8ef62e5aed5cda224a428d389e52b68768/install/runtime-payload/scripts/kata-deploy.sh#L202
```
        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-remote]
          runtime_type = "io.containerd.kata.v2"
          cri_handler = "cc"
          privileged_without_host_devices = true
          pod_annotations = ["io.katacontainers.*"]
```

I don't think any of per-pod Kata configuration options are usable for peer pods, so we can probably ignore them.
https://github.com/kata-containers/kata-containers/blob/main/docs/how-to/how-to-set-sandbox-config-kata.md

When we do not set `pod_annotations`, then all `io.katacontainers` annotations are ignored.

If It is OK, I'll raise an PR to fix the issue.

**Comment by yoheiueda** [comment]
The script I mentioned above will be deleted soon, and it turns out that I need to fix `kata-deploy.sh` in Kata Containers CCv0.

I raised an PR to fix this issue. https://github.com/kata-containers/kata-containers/pull/6970

**Comment by stevenhorsman** [comment]
This should have been resolved with #6970
